source/go-perun/app_tutorial/client.rst

```diff
@@ -0,0 +1,329 @@
+            lcp, ok := p.(*client.LedgerChannelProposal)
+            if !ok {
+                return nil, fmt.Errorf("Invalid proposal type: %T\n", p)
+            }
+
+            // Ensure the ledger channel proposal includes the expected app.
+            if !lcp.App.Def().Equals(c.app.Def()) {
+                return nil, fmt.Errorf("Invalid app type ")
+            }
+
+            // Check that we have the correct number of participants.
+            if lcp.NumPeers() != 2 { //TODO:go-perun rename NumPeers to NumParts, Peers to Participants anywhere where all parties are referred to
+                return nil, fmt.Errorf("Invalid number of participants: %d", lcp.NumPeers())
+            }
+
+            // Check that the channel has the expected assets.
+            err := channel.AssetsAssertEqual(lcp.InitBals.Assets, []channel.Asset{c.currency})
+            if err != nil {
+                return nil, fmt.Errorf("Invalid assets: %v\n", err)
+            }
+
+            // Check that the channel has the expected assets and funding balances.
+            const assetIdx, peerIdx = 0, 1
+            if err := channel.AssetsAssertEqual(lcp.InitBals.Assets, []channel.Asset{c.currency}); err != nil {
+                return nil, fmt.Errorf("Invalid assets: %v\n", err)
+            } else if lcp.FundingAgreement[assetIdx][peerIdx].Cmp(c.stake) != 0 {
+                return nil, fmt.Errorf("Invalid funding balance")
+            }
+            return lcp, nil
+        }()
+        if err != nil {
+            r.Reject(context.TODO(), err.Error()) //nolint:errcheck // It's OK if rejection fails.
+        }
+
+As in the payment channel example, we create the accept message, send it, and start the on-chain watcher.
+
+.. code-block:: go
+
+        // Create a channel accept message and send it.
+        accept := lcp.Accept(
+            c.account,                // The account we use in the channel.
+            client.WithRandomNonce(), // Our share of the channel nonce.
+        )
+        ch, err := r.Accept(context.TODO(), accept)
+        if err != nil {
+            fmt.Printf("Error accepting channel proposal: %v\n", err)
+            return
+        }
+
+        // Start the on-chain event watcher. It automatically handles disputes.
+        c.startWatching(ch)
+
+        c.channels <- newAppChannel(ch)
+    }
+
+.. _app_client_channel:
+
+App channel
+-----------
+
+We implement the type ``AppChannel`` that wraps a Perun channel and provides convenient functions to interact with the app channel.
+We put this functionality in ``client/channel.go``.
+
+.. code-block:: go
+
+    type AppChannel struct {
+        ch *client.Channel
+    }
+
+.. _app_client_channel_set:
+
+Set
+...
+We create ``Set``, which expects ``x`` and ``y`` that indicate where the client wants to put its symbol on the grid.
+
+We use ``channel.UpdateBy`` for conveniently proposing our desired update to the channel's ``state``.
```

Comment:
```suggestion
We use ``channel.UpdateBy`` for proposing our desired update to the channel's ``state``.
```

Try to omit words such as "conveniently", "easily" when you describe something that can be done by the reader. The reader can have its own opinion about whether its convenient or not, and it doesn't make it more convenient just because you name it.